diff --git a/lib/index.js b/lib/index.js
index e104b4fee8cf50921cf87152c014e4a2bba668dc..a8fff56f61a8f19f39691a0f8d03b498391b2aed 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -12,7 +12,6 @@ function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r
 (() => {
   var $util = $protobuf.util;
   if ($util.Long == null) {
-    console.log(`Patching Protobuf Long.js instance...`);
     $util.Long = _long.default;
     if ($protobuf.Reader._configure != null) {
       $protobuf.Reader._configure($protobuf.BufferReader);
diff --git a/src/index.js b/src/index.js
index aab22c7c39e6a40a022f25bd8e545e3ba0705905..f0072bef2ee3a9aaac482a818be2f46b10f29699 100644
--- a/src/index.js
+++ b/src/index.js
@@ -10,7 +10,6 @@ import * as $proto from "./proto.js";
     var $util = $protobuf.util;
 
     if ($util.Long == null) {
-        console.log(`Patching Protobuf Long.js instance...`);
         $util.Long = Long;
 
         if ($protobuf.Reader._configure != null) {
