diff --git a/cjs/base/Enum.js b/cjs/base/Enum.js
index b4839bb366f55e6cf9368fd6f5bfbd189c8e71e8..ea0ae0a1e526e80dd7637c7ad778e7d6a829f7ad 100644
--- a/cjs/base/Enum.js
+++ b/cjs/base/Enum.js
@@ -157,31 +157,32 @@ class Enum {
         let definition;
         // eslint-disable-next-line no-return-assign
         const setDefinition = (d) => definition = d;
-        return class extends Enum {
-            static {
-                const keys = Array.isArray(Types)
-                    ? Types
-                    : Object.keys(Types);
-                const count = keys.length;
-                const asKeys = new Array(count);
-                const isKeys = new Array(count);
-                for (let i = 0; i < count; i++) {
-                    const name = (0, util_1.stringPascalCase)(keys[i]);
-                    asKeys[i] = `as${name}`;
-                    isKeys[i] = `is${name}`;
-                }
-                (0, util_1.objectProperties)(this.prototype, isKeys, (_, i, self) => self.type === keys[i]);
-                (0, util_1.objectProperties)(this.prototype, asKeys, (k, i, self) => {
-                    if (self.type !== keys[i]) {
-                        throw new Error(`Cannot convert '${self.type}' via ${k}`);
-                    }
-                    return self.value;
-                });
-            }
+        const Cl = class extends Enum {
             constructor(registry, value, index) {
                 super(registry, Types, value, index, { definition, setDefinition });
             }
         };
+        (function () {
+            const keys = Array.isArray(Types)
+                ? Types
+                : Object.keys(Types);
+            const count = keys.length;
+            const asKeys = new Array(count);
+            const isKeys = new Array(count);
+            for (let i = 0; i < count; i++) {
+                const name = (0, util_1.stringPascalCase)(keys[i]);
+                asKeys[i] = `as${name}`;
+                isKeys[i] = `is${name}`;
+            }
+            (0, util_1.objectProperties)(Cl.prototype, isKeys, (_, i, self) => self.type === keys[i]);
+            (0, util_1.objectProperties)(Cl.prototype, asKeys, (k, i, self) => {
+                if (self.type !== keys[i]) {
+                    throw new Error(`Cannot convert '${self.type}' via ${k}`);
+                }
+                return self.value;
+            });
+        })();
+        return Cl;
     }
     /**
      * @description The length of the value when encoded as a Uint8Array
diff --git a/cjs/native/Set.js b/cjs/native/Set.js
index c080279f0e5c8cb1daaf74b15332ff4c239dc589..e76c7aeac625eaee41566ebbeef403ef16d7736d 100644
--- a/cjs/native/Set.js
+++ b/cjs/native/Set.js
@@ -82,20 +82,21 @@ class CodecSet extends Set {
         this.__internal__byteLength = bitLength / 8;
     }
     static with(values, bitLength) {
-        return class extends CodecSet {
-            static {
-                const keys = Object.keys(values);
-                const count = keys.length;
-                const isKeys = new Array(count);
-                for (let i = 0; i < count; i++) {
-                    isKeys[i] = `is${(0, util_1.stringPascalCase)(keys[i])}`;
-                }
-                (0, util_1.objectProperties)(this.prototype, isKeys, (_, i, self) => self.strings.includes(keys[i]));
-            }
+        const Cl = class extends CodecSet {
             constructor(registry, value) {
                 super(registry, values, value, bitLength);
             }
         };
+        (function () {
+            const keys = Object.keys(values);
+            const count = keys.length;
+            const isKeys = new Array(count);
+            for (let i = 0; i < count; i++) {
+                isKeys[i] = `is${(0, util_1.stringPascalCase)(keys[i])}`;
+            }
+            (0, util_1.objectProperties)(Cl.prototype, isKeys, (_, i, self) => self.strings.includes(keys[i]));
+        })();
+        return Cl;
     }
     /**
      * @description The length of the value when encoded as a Uint8Array
diff --git a/cjs/native/Struct.js b/cjs/native/Struct.js
index f4b4ea454472e1ecef3fb365e9280c6a87343c07..7126e763dbd55360175f825afef67c84fa47746f 100644
--- a/cjs/native/Struct.js
+++ b/cjs/native/Struct.js
@@ -98,15 +98,16 @@ class Struct extends Map {
         let definition;
         // eslint-disable-next-line no-return-assign
         const setDefinition = (d) => definition = d;
-        return class extends Struct {
-            static {
-                const keys = Object.keys(Types);
-                (0, util_1.objectProperties)(this.prototype, keys, (k, _, self) => self.get(k));
-            }
+        const Cl = class extends Struct {
             constructor(registry, value) {
                 super(registry, Types, value, jsonMap, { definition, setDefinition });
             }
         };
+        (function () {
+            const keys = Object.keys(Types);
+            (0, util_1.objectProperties)(Cl.prototype, keys, (k, _, self) => self.get(k));
+        })();
+        return Cl;
     }
     /**
      * @description The available keys for this struct
