diff --git a/node_modules/@shopify/react-native-performance/ios/ReactNativePerformance/ReactNativePerformance.m b/node_modules/@shopify/react-native-performance/ios/ReactNativePerformance/ReactNativePerformance.m
index 4852113..dbb4bb5 100644
--- a/node_modules/@shopify/react-native-performance/ios/ReactNativePerformance/ReactNativePerformance.m
+++ b/node_modules/@shopify/react-native-performance/ios/ReactNativePerformance/ReactNativePerformance.m
@@ -1,5 +1,5 @@
 #import "ReactNativePerformance.h"
-#import "ReactNativePerformance-Swift.h"
+#import <ReactNativePerformance/ReactNativePerformance-Swift.h>
 
 static NSTimeInterval startupTimestamp = -1.0;