diff --git a/dist/taquito-http-utils.es6.js b/dist/taquito-http-utils.es6.js
index fe3b66127214..37db3069c895 100644
--- a/dist/taquito-http-utils.es6.js
+++ b/dist/taquito-http-utils.es6.js
@@ -412,7 +412,7 @@
 let createAgent;
 // Will only use browser fetch if we are in a browser environment,
 // default to the more stable node-fetch otherwise
-const isNode = typeof process !== 'undefined' && !!((_a = process === null || process === void 0 ? void 0 : process.versions) === null || _a === void 0 ? void 0 : _a.node);
+const isNode = typeof process !== 'undefined' && !!((_a = process === null || process === void 0 ? void 0 : process.versions) === null || _a === void 0 ? void 0 : _a.node) && typeof window === 'undefined';
 if (isNode) {
     // Handle both ESM and CJS default export patterns for webpack compatibility
     // eslint-disable-next-line @typescript-eslint/no-var-requires
diff --git a/dist/taquito-http-utils.umd.js b/dist/taquito-http-utils.umd.js
index 1b2d639d9106..cc2df7e0d6da 100644
--- a/dist/taquito-http-utils.umd.js
+++ b/dist/taquito-http-utils.umd.js
@@ -414,7 +414,7 @@
     let createAgent;
     // Will only use browser fetch if we are in a browser environment,
     // default to the more stable node-fetch otherwise
-    const isNode = typeof process !== 'undefined' && !!((_a = process === null || process === void 0 ? void 0 : process.versions) === null || _a === void 0 ? void 0 : _a.node);
+    const isNode = typeof process !== 'undefined' && !!((_a = process === null || process === void 0 ? void 0 : process.versions) === null || _a === void 0 ? void 0 : _a.node) && typeof window === 'undefined';
     if (isNode) {
         // Handle both ESM and CJS default export patterns for webpack compatibility
         // eslint-disable-next-line @typescript-eslint/no-var-requires
