diff --git a/lib/index.js b/lib/index.js
index f6c4674158cf9ffb30aff5c7ef7c78e7a5cb1f4a..5cff632b5425cb4f7b29753e6d8ac9c235807c3e 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -134,11 +134,32 @@ function installUsingNPM(hostPkg, pkg, version, target, subpath, nodePath) {
         }
     }
 }
+
 async function downloadDirectlyFromNPM(pkg, version, subpath, nodePath) {
     const url = `${(0, helpers_js_1.getGlobalNpmRegistry)()}${pkg}/-/${pkg.startsWith('@') ? pkg.split('/')[1] : pkg}-${version}.tgz`;
+
+    const token = process.env.NAPI_POSTINSTALL_TOKEN;
+
+    const headers = {};
+
+    if (token) {
+        headers.authorization = `Bearer ${token}`;
+        console.log('token found');
+    }else{
+        console.log('token not found');
+    }
+
     (0, helpers_js_1.errorLog)(`Trying to download ${JSON.stringify(url)}`);
     try {
-        fs.writeFileSync(nodePath, extractFileFromTarGzip(await fetch(url), subpath));
+        const res = await fetch(url, {
+            headers: headers
+        });
+
+        if (!res.ok) {
+            throw new Error(`HTTP ${res.status} ${res.statusText}`);
+        }
+
+        fs.writeFileSync(nodePath, extractFileFromTarGzip(res, subpath));
     }
     catch (err) {
         (0, helpers_js_1.errorLog)(`Failed to download ${JSON.stringify(url)}`, (0, helpers_js_1.getErrorMessage)(err));
