diff --git a/android/src/main/cpp/CMakeLists.txt b/android/src/main/cpp/CMakeLists.txt
index 06ebc529e08e8f1f4cf44c360687cb34b3941679..5fc68204c185520e85ba3fdbc01c6725a6a04aa2 100644
--- a/android/src/main/cpp/CMakeLists.txt
+++ b/android/src/main/cpp/CMakeLists.txt
@@ -1,6 +1,6 @@
 cmake_minimum_required(VERSION 3.4.1)
 
-add_compile_options(-fvisibility=hidden -w)
+add_compile_options(-fvisibility=hidden -w -Wno-register)
 include_directories(${CMAKE_SOURCE_DIR}/scrypt)
 include_directories(${CMAKE_SOURCE_DIR}/../../../jni/include)
 
diff --git a/ios/RNFastCrypto.h b/ios/RNFastCrypto.h
index 0a8f93d97e9d873b919d14060575bbd6b3cfb35c..884ca905cc412ba366a852805d25b0ee11e57311 100644
--- a/ios/RNFastCrypto.h
+++ b/ios/RNFastCrypto.h
@@ -2,7 +2,7 @@
 #if __has_include("RCTBridgeModule.h")
 #import "RCTBridgeModule.h"
 #else
-#import <RCTBridgeModule.h>
+#import <React/RCTBridgeModule.h>
 #endif
 
 @interface RNFastCrypto : NSObject <RCTBridgeModule>
