diff --git a/android/src/main/java/com/appsamurai/storyly/reactnative/STStorylyManager.kt b/android/src/main/java/com/appsamurai/storyly/reactnative/STStorylyManager.kt
index 433a79f9bf84f692875d3e87bcfcb5e853222500..0b17edd83d3761f67ce726a235ba23bb053ee205 100644
--- a/android/src/main/java/com/appsamurai/storyly/reactnative/STStorylyManager.kt
+++ b/android/src/main/java/com/appsamurai/storyly/reactnative/STStorylyManager.kt
@@ -154,7 +154,7 @@ class STStorylyManager : ViewGroupManager<STStorylyView>() {
 
             COMMAND_REJECT_CART_CHANGE_CODE -> {
                 val responseId: String = args?.getString(0) ?: return
-                val failMessage: String = if (args.size() > 1) args.getString(1) else ""
+                val failMessage: String = if (args.size() > 1) args.getString(1)!! else ""
                 root.rejectCartChange(responseId, failMessage)
             }
 
diff --git a/android/src/main/java/com/appsamurai/storyly/reactnative/verticalFeed/STVerticalFeedManager.kt b/android/src/main/java/com/appsamurai/storyly/reactnative/verticalFeed/STVerticalFeedManager.kt
index a94cc8a82286fad7b81e10e2ea5da13cac4fe20a..97922dc5d8fe6a5ac3595d0f09317890c96665ce 100644
--- a/android/src/main/java/com/appsamurai/storyly/reactnative/verticalFeed/STVerticalFeedManager.kt
+++ b/android/src/main/java/com/appsamurai/storyly/reactnative/verticalFeed/STVerticalFeedManager.kt
@@ -151,7 +151,7 @@ class STVerticalFeedManager : ViewGroupManager<STVerticalFeedView>() {
 
             COMMAND_REJECT_CART_CHANGE_CODE -> {
                 val responseId: String = args?.getString(0) ?: return
-                val failMessage: String = if (args.size() > 1) args.getString(1) else ""
+                val failMessage: String = if (args.size() > 1) args.getString(1)!! else ""
                 root.rejectCartChange(responseId, failMessage)
             }
 
diff --git a/android/src/main/java/com/appsamurai/storyly/reactnative/verticalFeedBar/STVerticalFeedBarManager.kt b/android/src/main/java/com/appsamurai/storyly/reactnative/verticalFeedBar/STVerticalFeedBarManager.kt
index 5944eeaf03ab9ba26c70daa9af401a185c3109d8..d47bdbb34c2766f8bacb645b8bf07a701160c60b 100644
--- a/android/src/main/java/com/appsamurai/storyly/reactnative/verticalFeedBar/STVerticalFeedBarManager.kt
+++ b/android/src/main/java/com/appsamurai/storyly/reactnative/verticalFeedBar/STVerticalFeedBarManager.kt
@@ -151,7 +151,7 @@ class STVerticalFeedBarManager : ViewGroupManager<STVerticalFeedBarView>() {
 
             COMMAND_REJECT_CART_CHANGE_CODE -> {
                 val responseId: String = args?.getString(0) ?: return
-                val failMessage: String = if (args.size() > 1) args.getString(1) else ""
+                val failMessage: String = if (args.size() > 1) args.getString(1)!! else ""
                 root.rejectCartChange(responseId, failMessage)
             }
 
diff --git a/android/src/main/java/com/appsamurai/storyly/reactnative/verticalFeedPresenter/STVerticalFeedPresenterManager.kt b/android/src/main/java/com/appsamurai/storyly/reactnative/verticalFeedPresenter/STVerticalFeedPresenterManager.kt
index aa59ee13868fc949233994acf739e2ca54d060de..e388ca9d0e3ac849bd9631d277ed4c32bcf6f461 100644
--- a/android/src/main/java/com/appsamurai/storyly/reactnative/verticalFeedPresenter/STVerticalFeedPresenterManager.kt
+++ b/android/src/main/java/com/appsamurai/storyly/reactnative/verticalFeedPresenter/STVerticalFeedPresenterManager.kt
@@ -146,7 +146,7 @@ class STVerticalFeedPresenterManager : ViewGroupManager<STVerticalFeedPresenterV
 
             COMMAND_REJECT_CART_CHANGE_CODE -> {
                 val responseId: String = args?.getString(0) ?: return
-                val failMessage: String = if (args.size() > 1) args.getString(1) else ""
+                val failMessage: String = if (args.size() > 1) args.getString(1)!! else ""
                 root.rejectCartChange(responseId, failMessage)
             }
 
