diff --git a/native/dist/types.d.ts b/native/dist/types.d.ts
index 4fa2beb0ee73552025431962c4aab70d9ceccb33..cfca2671134cb746e436473fc54b15c3921ab714 100644
--- a/native/dist/types.d.ts
+++ b/native/dist/types.d.ts
@@ -16,7 +16,7 @@ export interface ExoticComponentWithDisplayName<P extends object = {}> extends R
 export type StyledComponentBrand = {
     readonly _sc: symbol;
 };
-export type BaseObject = {};
+export type BaseObject = { children?: React.ReactNode; style?: any; [key: string]: any };
 export type OmitNever<T> = {
     [K in keyof T as T[K] extends never ? never : K]: T[K];
 };
@@ -24,7 +24,7 @@ export type FastOmit<T extends object, U extends string | number | symbol> = {
     [K in keyof T as K extends U ? never : K]: T[K];
 };
 export type Runtime = 'web' | 'native';
-export type AnyComponent<P extends object = any> = ExoticComponentWithDisplayName<P> | React.ComponentType<P>;
+export type AnyComponent<P extends object = any> = ExoticComponentWithDisplayName<P> | React.ComponentType<P> | React.ForwardRefExoticComponent<any> | React.NamedExoticComponent<any> | React.ExoticComponent<any> | React.ComponentClass<any, any> | ((...args: any[]) => any) | (new (...args: any[]) => any);
 export type KnownTarget = SupportedHTMLElements | AnyComponent;
 export type WebTarget = string | KnownTarget;
 export type NativeTarget = AnyComponent;
