--- /Users/grenaudeau/Desktop/rnc/ios/RN/RNCameraManager.m	2019-02-25 12:07:29.000000000 +0100
+++ node_modules/react-native-camera/ios/RN/RNCameraManager.m	2019-02-25 13:36:21.000000000 +0100
@@ -386,16 +386,7 @@
 
 RCT_EXPORT_METHOD(checkRecordAudioAuthorizationStatus:(RCTPromiseResolveBlock)resolve
                   reject:(__unused RCTPromiseRejectBlock)reject) {
-#ifdef DEBUG
-    if ([[NSBundle mainBundle].infoDictionary objectForKey:@"NSMicrophoneUsageDescription"] == nil) {
-        RCTLogWarn(@"Checking audio permissions without having key 'NSMicrophoneUsageDescription' defined in your Info.plist. Audio Recording for your video files is therefore disabled. If you do not need audio on your recordings is is recommended to set the 'captureAudio' property on your component instance to 'false', otherwise you will have to add the key 'NSMicrophoneUsageDescription' to your Info.plist. If you do not your app will crash when being built in release mode. You can learn more about adding permissions here: https://stackoverflow.com/a/38498347/4202031");
-        resolve(@(NO));
-        return;
-    }
-#endif
-    [[AVAudioSession sharedInstance] requestRecordPermission:^(BOOL granted) {
-        resolve(@(granted));
-    }];
+  resolve(@(NO));
 }
 
 RCT_REMAP_METHOD(getAvailablePictureSizes,
