diff --git a/node_modules/react-native-gesture-handler/Swipeable.js b/Users/grenaudeau/Desktop/Swipeable.js
index ea88079a..f7189f6b 100644
--- a/node_modules/react-native-gesture-handler/Swipeable.js
+++ b/Users/grenaudeau/Desktop/Swipeable.js
@@ -26,7 +26,7 @@ export type PropType = {
   rightThreshold?: number,
   overshootLeft?: boolean,
   overshootRight?: boolean,
-  overshootFriction?: number,
+  overshootFriction: number,
   onSwipeableLeftOpen?: Function,
   onSwipeableRightOpen?: Function,
   onSwipeableOpen?: Function,